### ISSUE TITLE
Starting point for this plugin

### ISSUE THREAD

**Issue by acs** [lasteamlab/mcpythonmod#1]
Hi!

First of all, it is nice to see more people willing to help in the Raspberry PI support in new versions of Minecraft.

Have you started your plugin from https://github.com/zhuowei/RaspberryJuice/pull/84?

I have tested it, and it works. Probably it is the best starting point IMHO.

Cheers!

**Comment by lasteamlab** [comment]
Yes, I grabbed it from @MinecraftDawn and I've updated it so that the player class can handle player ids.  @sprouviere will be sending me his updates when he finishes this next round.  With that we will get entity classes that handle names instead of ids as well as things like "baby" and stairs and such.  I'm not going to mess with any of that code until I get his (he doesn't use github, so I'll merge things in slightly manually.)

His is the mcpi2 which won't fully work with this raspberryjuice2 yet, since it mcpi2 uses entity names and raspberryjuice2 doesn't, so we're a week out from things working well together.

**Comment by acs** [comment]
Cool guys! I have all these pieces in mind because this Raspberry Pi API es the main renderer in https://github.com/Voxelers/mcthings So I will be more than happy testing all your code (plugin and python lib).

Do you plan to adhere yourself to the current API or do you plan to extend it?

**Comment by lasteamlab** [comment]
My eventual goal is to add hooks to everything in bukkit, but we'll see how
that pans out.

**Comment by testmc76** [comment]
Hello! Good work. 
Сan i ask to add the getBlockWithData function at the beginning

**Comment by testmc76** [comment]
In CmdWorld.java on line 49:    } else if (command.equals("world.getBlockWithData")) { - need to remove "world"?

Even after the fix, it is not possible to save and restore the structure, maybe you can tell me how to do it?

**Comment by lasteamlab** [comment]
I'll look at this tomorrow probably.  I haven't played with that command
(this is a fork).  I'm not sure what you're trying to do with it, though.
This should just return the data for one block.

**Comment by testmc76** [comment]
> I'll look at this tomorrow probably. I haven't played with that command (this is a fork). I'm not sure what you're trying to do with it, though. This should just return the data for one block.

Thanks. I want to save structures from one world and build them in another world using a script. 
Example scripts:
[Save_load.zip](https://github.com/lasteamlab/RaspberryJuice2/files/4909180/Save_load.zip)

**Comment by lasteamlab** [comment]
The latest upload should allow you to save a structure, but with the move
to 1.16 the block data looks very different and that will affect the upload
file.  I'm going to spend some time tomorrow looking into figuring out how
best to load blocks with data in the format that get blocks with data
provides.  It is very verbose:
CraftBlockData{minecraft:dark_oak_fence_gate[facing=east,in_wall=false,open=false,powered=false]}
